src/main/java/seedu/TaskList.java

```diff
@@ -0,0 +1,152 @@
+    public TaskList(List<Task> t) {
+        this.lists = t;
+    }
+
+    /**
+     * Add a tasks in the lists of tasks
+     *
+     * @param task the Task class that is to be added
+     */
+    public void add(Task t) throws Exception {
+        try {
+            this.lists.add(t);
+        } catch(IllegalArgumentException e) {
+            throw new Exception("Unfortunately, " + e.getMessage());
+        } catch(Exception e) {
+            throw new Exception("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Returns the current task from the list
+     *
+     * @param index the position of the current task to be retrieved
+     * @returns a Task object of interest
+     */
+    public Task get(int index) {
+        return this.lists.get(index);
+    }
+
+    /**
+     * Retrieves task lists from a list of string from a file
+     *
+     * @param from source of the lists of strings
+     * @returns a TaskList of the current lists of tasks
+     */
+    public static TaskList getFromFile(List<String> from) throws Exception {
+        try {
+            List<Task> lists = new ArrayList<>();
+
+            for (String l : from) {
+                String[] chars = l.split("\\|");
+                if (chars[0].equals("ToDo")) {
+                    lists.add(new Task("todo" + " " + chars[2] ,"todo"));
+                } else if (chars[0].equals("Event")){
+                    String title = chars[2];
+                    String fr = chars[4].trim();
+                    String to = chars[3].trim();
+                    Task temp = new Task(title + " /from " + fr + " /to " + to, "event");
+                    if (chars[1].equals("true")) temp.isDone = true;
+                    lists.add(temp);
+                } else {
+                    String title = chars[2].trim();
+                    String to = chars[3].trim();
+                    Task temp = new Task(title + " /by " + to,"deadline");
+                    if (chars[1].equals("true")) temp.isDone = true;
+                    lists.add(temp);
+                }
+            }
+            return new TaskList(lists);
+        } catch(Exception e) {
+            throw new Error("Parsing error");
+        }
+
+    }
+
+    /**
+     * Encodes TaskList to a list of string to be used in a file
+     *
+     * @param tasks the list of Tasks objects
+     * @returns a String List to be stored in a file
+     */
+    public static List<String> encode(TaskList tasks) {
+        int curr = 1;
+        List<String> arrays = new ArrayList<>();
+        for (int i = 0; i < tasks.getLen(); i++) {
+            if (tasks.get(i) != null) {
+                Task t = tasks.get(i);
+                String start = !t.start.isEmpty() ? t.start + "|" : "";
+                String end = !t.end.isEmpty()?  t.end + "|" : "";
+                arrays.add(t.category.toString() + "|" + (t.isDone ? "true" : "false") + "|" + t.title + "|" + end + start);
```

Comment:
Line is too long, you may want to consider breaking the line (with appropriate indentation).